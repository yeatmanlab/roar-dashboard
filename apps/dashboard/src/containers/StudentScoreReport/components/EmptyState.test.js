import { describe, it, expect, beforeEach, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';
import EmptyStateComponent from './EmptyState.vue';

// Mock router
const mockRouter = {
  go: vi.fn(),
};

describe('EmptyState.vue', () => {
  let i18n;

  beforeEach(() => {
    vi.clearAllMocks();

    i18n = createI18n({
      legacy: false,
      locale: 'en',
      messages: {
        en: {
          scoreReports: {
            stillWorking: '{firstName} is still working!',
            needOneComplete: '{firstName} needs to complete at least one assessment.',
          },
        },
      },
    });
  });

  describe('snapshots', () => {
    it('renders with student first name', () => {
      const wrapper = mount(EmptyStateComponent, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
          mocks: {
            $router: mockRouter,
          },
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders with different student name', () => {
      const wrapper = mount(EmptyStateComponent, {
        props: {
          studentFirstName: 'Sarah',
        },
        global: {
          plugins: [i18n],
          mocks: {
            $router: mockRouter,
          },
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('component behavior', () => {
    it('displays student first name in title', () => {
      const wrapper = mount(EmptyStateComponent, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
          mocks: {
            $router: mockRouter,
          },
        },
      });

      expect(wrapper.text()).toContain('John');
      expect(wrapper.text()).toContain('is still working!');
    });

    it('displays student first name in description', () => {
      const wrapper = mount(EmptyStateComponent, {
        props: {
          studentFirstName: 'Sarah',
        },
        global: {
          plugins: [i18n],
          mocks: {
            $router: mockRouter,
          },
        },
      });

      expect(wrapper.text()).toContain('Sarah');
      expect(wrapper.text()).toContain('needs to complete at least one assessment');
    });

    it('calls router.go(-1) when back button is clicked', async () => {
      const wrapper = mount(EmptyStateComponent, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
          mocks: {
            $router: mockRouter,
          },
        },
      });

      const backButton = wrapper.find('button');
      await backButton.trigger('click');

      expect(mockRouter.go).toHaveBeenCalledWith(-1);
    });

    it('renders back button with correct text', () => {
      const wrapper = mount(EmptyStateComponent, {
        props: {
          studentFirstName: 'John',
        },
        global: {
          plugins: [i18n],
          mocks: {
            $router: mockRouter,
          },
        },
      });

      expect(wrapper.text()).toContain('Back to overview');
    });
  });
});
