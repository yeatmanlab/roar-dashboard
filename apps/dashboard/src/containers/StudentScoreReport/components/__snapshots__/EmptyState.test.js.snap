// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyState.vue > snapshots > renders with different student name 1`] = `
"<div class="px-4 py-8 mt-2 text-center bg-white rounded border-2 border border-gray-100">
  <h2 class="font-bold text-gray-600 text-md">Sarah is still working!</h2>
  <p class="text-gray-400">Sarah needs to complete at least one assessment.</p>
  <div class="mt-5">
    <!-- TODO: Fix routing to always go back to overview --><button class="p-button p-component p-button-secondary" data-p="" type="button" aria-label="Back to overview " data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" pc0="" data-pc-section="root"><span class="p-button-icon p-button-icon-left pi pi-arrow-left" data-p="left" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label" data-p="">Back to overview </span>
      <!---->
    </button>
  </div>
</div>"
`;

exports[`EmptyState.vue > snapshots > renders with student first name 1`] = `
"<div class="px-4 py-8 mt-2 text-center bg-white rounded border-2 border border-gray-100">
  <h2 class="font-bold text-gray-600 text-md">John is still working!</h2>
  <p class="text-gray-400">John needs to complete at least one assessment.</p>
  <div class="mt-5">
    <!-- TODO: Fix routing to always go back to overview --><button class="p-button p-component p-button-secondary" data-p="" type="button" aria-label="Back to overview " data-pc-name="button" data-p-disabled="false" data-p-severity="secondary" data-html2canvas-ignore="true" pc0="" data-pc-section="root"><span class="p-button-icon p-button-icon-left pi pi-arrow-left" data-p="left" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label" data-p="">Back to overview </span>
      <!---->
    </button>
  </div>
</div>"
`;
